### PR TITLE
#166029408 implementing Feedback on api

### PR DESCRIPTION
**Pull request by danielufeli**
### What does this PR do?
implementing Feedback on API

### Description of Task to be completed?
- Refactored error handling
- Created a  middleware for error and moved all validations to routes
- Refactored users controlled
- Added a middleware for user object
- Refactored loans and repayment controllers
- Installed morgan and Winston to better log errors
- installed pg module for PostgreSQL
- Create sql queries and database connection

### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2326698

Finishes: #166029408

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,3 @@
-jwtPrivateKey=mySecureKey
\ No newline at end of file
+DATABASE_URL=postgres://daniel:Dominion2019@localhost:5432/quickcreditdb
+jwtPrivateKey=mySecureKey
+NODE_ENV=dev
```
Comment (V-84) [inline .env]: please enter newline here

File: server/controllers/loans.js [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-import validateLoanStatus from '../helpers/validation/loanstatus';
-import validateAmount from '../helpers/validation/amount';
 import repaymentHistory from '../helpers/repaymenthistory';
 
 
 class loansController {
-  static async createLoan(req, res) {
-    const { error } = validateLoan(req.body);
-    if (error) return res.status(400).json(error.details[0].message);
+  static createLoan(req, res) {
```
Comment (V-84) [inline server/controllers/loans.js:12]: this function exceeds 25lines... Please ensure it doesn't..

File: server/routes/adminusers.js [hunk 3]
```diff
@@ -3,13 +3,16 @@ import express from 'express';
 import adminUserController from '../controllers/adminusers';
 import auth from '../helpers/auth';
 import checkIsAdmin from '../middleware/isAdmin';
+import allValidator from '../middleware/allValidator';
+import getUserId from '../middleware/getUserId';
+import validateUserStatus from '../helpers/validation/status';
 
 const router = express.Router();
 
 // @route Get api/v1/users/test
```
Comment (V-84) [inline server/routes/adminusers.js]: This is not the correct JSDocs comment

File: server/routes/users.js [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 import userController from '../controllers/users';
+import allValidator from '../middleware/allValidator';
+import validateUser from '../helpers/validation/users';
+import checkUser from '../helpers/currentUser';
+import validateSignin from '../helpers/validation/signin';
 
 const router = express.Router();
 
 // @route Get api/v1/users/test
 // @desc Test post route
```
Comment (V-84) [inline server/routes/users.js:12]: same here

**Review by V-84 — CHANGES_REQUESTED** [review]
attend to the request

File: .env (hunk 1 above)
Comment (danielufeli) [reply .env]: ok resolved it

File: server/controllers/loans.js (hunk 2 above)
Comment (danielufeli) [reply server/controllers/loans.js:12]: ok resolved it

File: server/routes/adminusers.js (hunk 3 above)
Comment (danielufeli) [reply server/routes/adminusers.js]: ok resolved it

**Pull request merged**